sign_verify_file_ssh.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+
+# Usage message
+usage() {
+    echo "Usage: $0 sign <file> <private_key> | verify <file> <public_key> [signature_file]"
+    exit 1
+}
+
+# Ensure at least two arguments are provided
+if [ "$#" -lt 3 ]; then
+    usage
+fi
+
+MODE="$1"
+FILE_TO_SIGN="$2"
+
+# Ensure the file exists
+if [ ! -f "$FILE_TO_SIGN" ]; then
+    echo "Error: File '$FILE_TO_SIGN' not found."
+    exit 1
+fi
+
+# Function to securely convert PEM to OpenSSH if needed
+convert_pem_to_ssh() {
+    local key_file="$1"
+    local output_file="$2"
+
+    if grep -q "BEGIN RSA PRIVATE KEY" "$key_file" || grep -q "BEGIN OPENSSH PRIVATE KEY" "$key_file"; then
+        # Ensure cleanup on exit
+        trap 'rm -f "$output_file" "$output_file.pub"' EXIT
```

Comment:
That (`trap ...`) looks a bit fancy to me. First time we use it in any script in EESSI? Maybe not bad ... but then other scripts should be modernised too.

Reply:
I added it because I don't want to leave these files lying around even if the script is interrupted